lib/Document/StaticPageGenerator.php

```diff
@@ -28,8 +28,11 @@
 
 class StaticPageGenerator
 {
-    public function __construct(protected DocumentRenderer $documentRenderer, private LockFactory $lockFactory, protected SystemSettingsConfig $settingsConfig)
-    {
+    public function __construct(
+        protected DocumentRendererInterface $documentRenderer,
```

Comment:
That's unfortunately a BC-break, as this class is not `@internal` 😬

Reply:
Why? When someone overwrite this class it can still have the old cunstruct and it works.

Reply:
That's right, covariance does not affect constructors.

Reply:
👍 Thanks for clarifying guys - was not entirely sure about that one 😊